test/conftest.py

```diff
@@ -4,25 +4,46 @@
 from uuid import uuid4
 from contextlib import contextmanager
 from pathlib import Path
+from flask.testing import FlaskClient
 
 from multinet import create_app
 from multinet.db.models.workspace import Workspace
-from multinet.db.models.user import User, MULTINET_COOKIE
+from multinet.db.models.user import User
+from multinet.auth.util import create_login_token, encode_auth_token
 
 from typing import Generator, Tuple
 
 
 @contextmanager
-def login(user: User, server) -> Generator[None, None, None]:
-    """Perform server actions under a user login."""
+def login(user: User, server: FlaskClient) -> Generator[None, None, None]:
+    """
+    Perform server actions under a user login.
+
+    This is done by monkey-patching the FlaskClient `open` method, to include the
+    proper auth headers. The `open` method is the underlying method used by `get`,
+    `post`, `put`, etc.
+    """
 
-    with server.session_transaction() as session:
-        session[MULTINET_COOKIE] = user.multinet.session
+    # To create a user login token, we need the app context,
+    # since the encoding function uses flask_secret_key
+    with server.application.app_context():
+        login_token = create_login_token(user.multinet.session)
+        encoded_login_token = encode_auth_token(login_token)
 
-    yield None
+    default_open = server.open
 
-    with server.session_transaction() as session:
-        del session[MULTINET_COOKIE]
+    def logged_in_open(self, *args, **kwargs):
+        return default_open(
+            self,
+            *args,
+            **kwargs,
+            headers={"Authorization": f"Bearer {encoded_login_token}"},
+        )
+
+    # Ignore flake8, since it complains about the monkey-patching
```

Comment:
Interesting! What does it complain about, specifically?

Reply:
Two errors occur for `server.open = logged_in_open`.
On `server.open`, I get the following:
```
Cannot assign to a method
```

On `= logged_in_open`, I get the following:
```
Incompatible types in assignment (expression has type "Callable[[Any, VarArg(Any), KwArg(Any)], Any]", variable has type overloaded function)
```

I don't know if there'd be any way for me to resolve the first one properly, and I'm not sure how I'd fix the second one. Since this is just used in testing, and I can verify it works, I think the ignores here are justified.

Reply:
> Since this is just used in testing, and I can verify it works, I think the ignores here are justified.

I agree.

If you are curious enough, I'd like you to (locally) try [this workaround](https://github.com/python/mypy/issues/708#issuecomment-109783888) (just to see if it works). I don't think jumping through that hoop just to satisfy `mypy` in this particular case would be worthwhile, but I do wonder if it would actually work.

Reply:
> If you are curious enough, I'd like you to (locally) try [this workaround](https://github.com/python/mypy/issues/708#issuecomment-109783888) (just to see if it works). I don't think jumping through that hoop just to satisfy `mypy` in this particular case would be worthwhile, but I do wonder if it would actually work.

I tried this and it doesn't really seem to work, it also seems to be a slightly different case than ours.

Reply:
Cool. Thanks for checking!